e2e/aws_disk.go

```diff
@@ -0,0 +1,228 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type nodeDisks struct {
+	disks []disk
+	node  corev1.Node
+}
+
+type disk struct {
+	size int
+}
+
+const (
+	awsPurposeTag       = "lvm-e2etest"
+	labelNodeRoleWorker = "node-role.kubernetes.io/worker"
+)
+
+// getAWSNodeInfo returns instanceID, region, zone, error
+func getAWSNodeInfo(node corev1.Node) (string, string, string, error) {
+	var instanceID, region, zone string
+	// providerID looks like: aws:///us-east-2a/i-02d314dea14ed4efb
+	if !strings.HasPrefix(node.Spec.ProviderID, "aws://") {
+		return "", "", "", fmt.Errorf("not an aws based node")
+	}
+	split := strings.Split(node.Spec.ProviderID, "/")
+	instanceID = split[len(split)-1]
+	zone = split[len(split)-2]
+	region = zone[:len(zone)-1]
+	return instanceID, region, zone, nil
+}
+
+// this assumes that the device spaces /dev/sd[h-z] are available on the node
+// do not provide more than 20 disksizes
+// do not use more than once per node
+// this function is async
+func createAndAttachAWSVolumes(t *testing.T, ec2Client *ec2.EC2, namespace string, nodeEnv []nodeDisks) {
+	for _, nodeEntry := range nodeEnv {
+		go createAndAttachAWSVolumesForNode(t, nodeEntry, ec2Client)
+	}
+}
+
+func createAndAttachAWSVolumesForNode(t *testing.T, nodeEntry nodeDisks, ec2Client *ec2.EC2) {
+	node := nodeEntry.node
+	volumes := make([]*ec2.Volume, len(nodeEntry.disks))
+	volumeLetters := []string{"g", "h"}
+	volumeIDs := make([]*string, 0)
+	instanceID, _, zone, err := getAWSNodeInfo(node)
+	if err != nil {
+		t.Errorf("failed to create and attach aws disks for node %q: %+v", nodeEntry.node.Name, err)
```

Comment:
This will cause the tests to fail when run against our baremetal clusters.

Reply:
When would we want to run the e2e tests on baremetal clusters? 
Only for testing locally on SNO?

Reply:
Yes, the e2e-tests are meant to be run against local clusters when required as well.